packages/xstate-store/src/atom.ts

```diff
@@ -39,6 +40,30 @@ const [
 
 let activeSub: Subscriber | undefined = undefined;
 
+type AsyncAtomState<Data, Error = unknown> =
+  | { status: 'pending' }
+  | { status: 'fulfilled'; data: Data }
+  | { status: 'rejected'; error: Error };
+
+export function createAsyncAtom<T>(
```

Comment:
it would be good to have a test with a single async atom and multiple reads from it/subscribers, it should check that `getValue` is only called once (I think that's the expected result, right? 🤔 )